wgpu-hal/src/dx11/adapter.rs

```diff
@@ -203,6 +205,19 @@ impl super::Adapter {
         let max_compute_workgroups_per_dimension =
             d3d11::D3D11_CS_DISPATCH_MAX_THREAD_GROUPS_PER_DIMENSION;
 
+        let max_sampled_textures_per_shader_stage = max_sampled_textures;
+        let max_samplers_per_shader_stage = max_samplers;
+        let max_storage_buffers_per_shader_stage = max_uavs;
+        let max_storage_textures_per_shader_stage = max_uavs;
+        let max_uniform_buffers_per_shader_stage = max_constant_buffers;
+
+        let max_bindings_per_bind_group = (max_sampled_textures_per_shader_stage
+            + max_samplers_per_shader_stage
+            + max_storage_buffers_per_shader_stage
+            + max_storage_textures_per_shader_stage
+            + max_uniform_buffers_per_shader_stage)
+            * MAX_SHADER_STAGES_PER_PIPELINE;
```

Comment:
I don't think this is correct. `maxBindingsPerBindGroup`'s purpose is to limit the maximum index used for bind group bindings, because in some implementations (wgpu with the vulkan backend for example) binding descriptors end up being allocated in a dense array, the size of which is the biggest index in the group, and with some vulkan drivers that was causing some stability/safety issues with large indices.

The limit was first introduced in the spec to be 640, and it has apparently been changed to 1000 since.

So the fix should involve updating the limit to 1000 and making it work consistently over the various backends.